data-model.d.ts

```diff
@@ -204,33 +272,59 @@ export interface BaseQuad {
 /**
  * An RDF quad, containing the subject, predicate, object and graph terms.
  */
-export interface Quad extends BaseQuad {
+ export interface Quad<
+    RdfMode extends AllowedRdfMode=RdfMode_star,
```

Comment:
By constraining to `AllowedRdfMode` you essentially close the path for 3rd party modes, contradicting (?) your comment about "easyRDF"

Reply:
I would also be in favor of a mechanism that enables developers to add other modes.
The approach from @tpluscode seems to enable this, so perhaps a combination of both approaches could be applied?